lib/src/register/register_page.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../l10n/tr.dart';
+import '../home/home_page.dart';
+import '../register/cubits/register_cubit.dart';
+
+class RegisterPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Center(
+        child: BlocListener<RegisterCubit, RegisterState>(
+          listener: (context, state) {
+            if (state is RegisterLoading) {
+              Text(tr(context).common_loading);
+            } else if (state is RegisterSucess) {
+              Navigator.push(
+                context,
+                MaterialPageRoute(builder: (context) => HomePage()),
+              );
+            } else {
+              Text(tr(context).error_generic);
+            }
+          },
+          child: ElevatedButton(
+            onPressed: () {
+              context.read<RegisterCubit>().register();
+            },
+            child: Text(tr(context).register),
```

Comment:
Disable button and show loading indication when it's fetching, to avoid multiple fetches.

Reply:
This better be postponed to be handled when adding the registration form as we can do it using Formz state `state.status`, what do you think?

Reply:
We can use `RegisterLoading` for now. The home page loaded twice because of this when I ran the app.